src/centralstore/incremental/NodeBlock.cpp

```diff
@@ -0,0 +1,261 @@
+        if (!node) {
+            throw "Error creating node in the relation";
+        }
+        edges.push_back(*node);
+    }
+    return edges;
+}
+
+std::map<std::string, char*> NodeBlock::getAllProperties() {
+    std::map<std::string, char*> allProperties;
+    PropertyLink* current = this->getPropertyHead();
+    while (current) {
+        allProperties.insert({current->name, current->value});
+        PropertyLink* _temp = current->next();
+        delete current;  // To prevent memory leaks
+        current = _temp;
+    }
+    delete current;
+    return allProperties;
+}
+
+NodeBlock* NodeBlock::get(unsigned int blockAddress) {
+    NodeBlock* nodeBlockPointer = NULL;
+    NodeBlock::nodesDB->seekg(blockAddress);
+    unsigned int edgeRef;
+    unsigned int propRef;
+    char usageBlock;
+    char label[6];
+    std::string id;
+
+    if (!NodeBlock::nodesDB->get(usageBlock)) {
+        node_block_logger.error("Error while reading usage data from block " + std::to_string(blockAddress));
+    }
+
+    if (!NodeBlock::nodesDB->read(reinterpret_cast<char*>(&edgeRef), sizeof(unsigned int))) {
+        node_block_logger.error("Error while reading edge reference data from block " + std::to_string(blockAddress));
+    }
+
+    if (!NodeBlock::nodesDB->read(reinterpret_cast<char*>(&propRef), sizeof(unsigned int))) {
+        node_block_logger.error("Error while reading prop reference data from block " + std::to_string(blockAddress));
+    }
+
+    if (!NodeBlock::nodesDB->read(&label[0], 6)) {
+        node_block_logger.error("Error while reading label data from block " + std::to_string(blockAddress));
+    }
+    bool usage = usageBlock == '\1';
+    node_block_logger.debug("Label = "+ std::string(label));
+    node_block_logger.debug("Label = " + std::string(label));
+    node_block_logger.debug("Length of label = " + std::to_string(strlen(label)));
+    node_block_logger.debug("edgeRef = " + std::to_string(edgeRef));
+    if (strlen(label) != 0) {
+        id = std::string(label);
+    }
+    nodeBlockPointer = new NodeBlock(id, blockAddress, propRef, edgeRef, label, usage);
+    if (nodeBlockPointer->id.length() == 0) {  // if label not found in node block look in the properties
+        std::map<std::string, char*> props = nodeBlockPointer->getAllProperties();
+        if (props["label"]) {
+            nodeBlockPointer->id = props["label"];
+        } else {
+            throw "Could not find node ID/Label for node with block address = " +
+                std::to_string(nodeBlockPointer->addr);
+        }
+    }
+    node_block_logger.debug("Edge ref = "+std::to_string(nodeBlockPointer->edgeRef));
+    if (nodeBlockPointer->edgeRef % RelationBlock::BLOCK_SIZE != 0) {
+        throw "Exception: Invalid edge reference address = " + nodeBlockPointer->edgeRef;
+    }
+    return nodeBlockPointer;
+}
+
+PropertyLink* NodeBlock::getPropertyHead() { return PropertyLink::get(this->propRef); }
+const unsigned long NodeBlock::BLOCK_SIZE = 15;
```

Comment:
Shall we move this constant declaration to `NodeBlock.h` header file?

Reply:
Fixed in https://github.com/miyurud/jasminegraph/pull/115/commits/1d404d7d8fc79e4d908dcc39667a33b78beeaa26

Reply:
Done